go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -54,14 +55,23 @@ func newEgressFirewallRule(rawEgressFirewallRule egressfirewallapi.EgressFirewal
 		access: rawEgressFirewallRule.Type,
 	}
 
-	_, _, err := net.ParseCIDR(rawEgressFirewallRule.To.CIDRSelector)
-	if err != nil {
-		return nil, err
-	}
-	efr.to.cidrSelector = rawEgressFirewallRule.To.CIDRSelector
+	if rawEgressFirewallRule.To.DNSName != "" {
+		efr.to.dnsName = rawEgressFirewallRule.To.DNSName
+	} else {
 
+		_, _, err := net.ParseCIDR(rawEgressFirewallRule.To.CIDRSelector)
+		if err != nil {
+			return nil, err
+		}
+		efr.to.cidrSelector = rawEgressFirewallRule.To.CIDRSelector
+	}
 	efr.ports = rawEgressFirewallRule.Ports
 
+	if efr.access == egressfirewallapi.EgressFirewallRuleDeny && len(efr.to.dnsName) != 0 {
+		return nil, fmt.Errorf("do not use dnsName as part of a deny rule this is insecure")
```

Comment:
Wording suggestion: "DNSName not allowed in deny rules"

Reply:
done